### PR TITLE
refactor!: unify `SpacersGenerator` and `PaddingsGenerator` in `ReferenceThemeClassGenerator`

### LINKED ISSUE #43: fix: number based classes (paddings, spacers, ...) use wrong number class types
**Description**

The named constructor of the number class works like expected:
```Dart
  const NumbersSemantic.darkMode()
      : numberWith2Aliases = 17.0,
        numberWithAlias = 17.0,
        number = 12.0;
```

The spacer class tries to import a wrongly named class:
```Dart
@immutable
class SpacersSemanticSpacer {
  const SpacersSemanticSpacer(this.spacersSemantic);

  final SpacersSemantic spacersSemantic; // <- that is the problem
...
```

### PR DESCRIPTION
**Pull request by timcreatedit**
## Description

A new superclass called `ReferenceThemeClassGenerator` abstracts most of the functionality for numbers-based generators.

fixes #43 

## Checklist

- [X] My PR title is in the style of [conventional commits](https://www.conventionalcommits.org/)
- [X] All public facing APIs are documented with [dartdoc](https://dart.dev/guides/language/effective-dart/documentation)
- [X] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/data/generators/reference_generators/reference_theme_class_generator.dart [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  /// The [Reference] to the type of [fromClass].
+  final Reference fromClass;
+
+  /// The fields in [fromClass] that will be used to generate the getters.
+  ///
+  /// In a numbers class for example, these could be `["s", "m", "l"]`.
+  final List<String> valueFields;
+
+  /// The name of the field in the generated class that holds the reference to
+  /// the fr.
```
Comment (JesperBllnbm) [inline lib/src/data/generators/reference_generators/reference_theme_class_generator.dart:35]: to the fr.?

File: lib/src/data/generators/theme_extension_generators/mode_theme_extension_generator.dart [hunk 2]
```diff
@@ -155,14 +141,14 @@ abstract class ValuesByModeThemeExtensionGenerator<T>
         ..fields.addAll(
           _getClassFields(
             nameList: parameterNames,
-            nullableSymbolReference: nullableSymbolReference,
+            nullableSymbolReference: symbolReference.toNullable,
```
Comment (JesperBllnbm) [inline lib/src/data/generators/theme_extension_generators/mode_theme_extension_generator.dart:144]: Could you find out why we decided to use nullable here? Or did you keep it to make it future proof for unresolved variables?

**Review by JesperBllnbm — APPROVED** [review]
LGTM!

File: lib/src/data/generators/theme_extension_generators/mode_theme_extension_generator.dart (hunk 2 above)
Comment (timcreatedit) [reply lib/src/data/generators/theme_extension_generators/mode_theme_extension_generator.dart:144]: No, but I would fix this in another PR once the whole unresolved thing is figured out :)

File: lib/src/data/generators/reference_generators/reference_theme_class_generator.dart (hunk 1 above)
Comment (timcreatedit) [reply lib/src/data/generators/reference_generators/reference_theme_class_generator.dart:35]: fr.

**Pull request merged**